src/Parameters.jl

```diff
@@ -218,29 +223,51 @@ A(3, 4)
 
 julia> b = reconstruct(a, b=99)
 A(3, 99)
+
+julia> struct B{T}
+          a::T
+          b
+       end
+
+julia> a = B(sin, 1)
+B{typeof(sin)}(sin, 1)
+
+julia> reconstruct(B, a, a=cos) # note reconstruct(a, a=cos) errors!
```

Comment:
and here? `julia> reconstruct(B, x, a=cos) # note reconstruct(x, a=cos) errors!`